### PR TITLE
Restructure project to enable custom mavlink bindings generation

### PR DESCRIPTION
**Pull request by GrahamDennis**
The goal of this PR is to enable generating Rust MAVLink bindings from custom MAVLink dialects in new crates.  This is done by splitting the project into three crates:

* mavlink-core: core MAVLink types including TCP, UDP and serial connections
* mavlink-bindgen: a library & CLI for generating Rust bindings for MAVLink dialects
* mavlink: generated MAVLink bindings for the MAVLink dialects in the mavlink/mavlink repo

mavlink-bindgen can be used to create MAVLink bindings for Rust, and this is used from `build.rs` in the `mavlink` crate to create bindings from the standard MAVLink dialects in https://github.com/mavlink/mavlink.

The `mavlink` crate is intended to remain API-compatible with the previous structure.

### CODE REVIEW

**Comment by GrahamDennis** [comment]
This now passes all checks in my fork, so is ready for review @patrickelectric . The biggest change here is refactoring the crate into 3 crates in a single cargo workspace (which I think is required to supporting custom dialects). I'm happy to discuss alternative approaches or structures of a cargo workspace. Thanks!

**Comment by patrickelectric** [comment]
That looks awesome, thanks @GrahamDennis. I'll review asap.

File: mavlink/build/main.rs [hunk 1]
```diff
@@ -0,0 +1,54 @@
+#![recursion_limit = "256"]
+
+use mavlink_bindgen;
```
Comment (patrickelectric) [inline mavlink/build/main.rs]: is this import necessary ?

**Review by patrickelectric — CHANGES_REQUESTED** [review]
There is a mavlink-embedded binary lost in the PR.

File: mavlink/build/main.rs (hunk 1 above)
Comment (GrahamDennis) [reply mavlink/build/main.rs]: No, removed.

**Review by GrahamDennis — COMMENTED** [review]
I've also removed the accidentally-commited mavlink-embedded binary in the root directory.

**Review by patrickelectric — APPROVED** [review]

**Comment by booo** [comment]
This change broke two links in the main README to the examples.

**Pull request merged**